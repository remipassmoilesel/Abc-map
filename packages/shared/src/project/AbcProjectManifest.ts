/**
 * Copyright © 2023 Rémi Pace.
 * This file is part of Abc-Map.
 *
 * Abc-Map is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as
 * published by the Free Software Foundation, either version 3 of
 * the License, or (at your option) any later version.
 *
 * Abc-Map is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General
 * Public License along with Abc-Map. If not, see <https://www.gnu.org/licenses/>.
 */

import { AbcLayer } from './layer';
import { AbcLayout } from './layout';
import { AbcView } from './AbcView';
import { AbcSharedView } from './shared-views';

export interface AbcProjectManifest {
  // Metadata: id, version, etc ...
  metadata: AbcProjectMetadata;
  // Main view for edition
  view: AbcView;
  // Layers: raster, geometries, ...
  layers: AbcLayer[];
  // Layouts for PDF exports
  layouts: {
    abcMapAttributionsEnabled: boolean;
    list: AbcLayout[];
  };
  // Views publicly accessible
  sharedViews: {
    // If true, map is fullscreen
    fullscreen: boolean;
    // If not fullscreen, these dimensions are used
    mapDimensions: { width: number; height: number };
    // List of views
    list: AbcSharedView[];
  };
}

export interface AbcProjectMetadata {
  id: string;
  version: string;
  name: string;
  // True if project can be accessed publicly
  public: boolean;
}
