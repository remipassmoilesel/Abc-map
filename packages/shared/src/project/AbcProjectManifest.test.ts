/**
 * Copyright © 2021 Rémi Pace.
 * This file is part of Abc-Map.
 *
 * Abc-Map is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as
 * published by the Free Software Foundation, either version 3 of
 * the License, or (at your option) any later version.
 *
 * Abc-Map is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General
 * Public License along with Abc-Map. If not, see <https://www.gnu.org/licenses/>.
 */

import { AbcProjectManifest } from './AbcProjectManifest';

/**
 * If this test fail, you should write a migration script then adapt it
 */
describe('AbcProjectManifest', () => {
  it('should not change without migration', () => {
    const actual: AbcProjectManifest = {
      metadata: {
        id: 'test-project-id',
        version: '0.1',
        name: 'Test project',
        public: false,
      },
      layers: [],
      layouts: {
        list: [],
        abcMapAttributionsEnabled: true,
      },
      view: {
        center: [1, 2],
        resolution: 1000,
        projection: { name: 'EPSG:4326' },
        rotation: 0,
      },
      sharedViews: {
        fullscreen: false,
        mapDimensions: {
          width: 800,
          height: 600,
        },
        list: [],
      },
    };

    expect(actual).toMatchSnapshot();
  });
});
