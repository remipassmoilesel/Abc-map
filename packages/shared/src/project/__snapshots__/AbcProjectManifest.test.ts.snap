// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AbcProjectManifest should not change without migration 1`] = `
Object {
  "layers": Array [],
  "layouts": Array [],
  "metadata": Object {
    "containsCredentials": false,
    "id": "test-project-id",
    "name": "Test project",
    "public": false,
    "version": "0.1",
  },
  "sharedViews": Object {
    "fullscreen": false,
    "list": Array [],
    "mapDimensions": Object {
      "height": 600,
      "width": 800,
    },
  },
  "view": Object {
    "center": Array [
      1,
      2,
    ],
    "projection": Object {
      "name": "EPSG:4326",
    },
    "resolution": 1000,
    "rotation": 0,
  },
}
`;
